adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/HttpClientBuilder.java

```diff
@@ -376,6 +376,14 @@ public HttpClient build(AdapterHttpClientConfig adapterConfig) {
 
         configureProxyForAuthServerIfProvided(adapterConfig);
 
+        if (socketTimeout == -1 && adapterConfig.getSocketTimeout() != -1) {
```

Comment:
```suggestion
        if (socketTimeout == -1 && adapterConfig.getSocketTimeout() > 0) {
```

Reply:
Probably it's better condition. I assume, nobody wants to set socket timeout to 0; therefore in this case, the default value should be used.

Reply:
It seems that zero is the default value which corresponds to infinite timeout.